misc/multiaddr/src/lib.rs

```diff
@@ -269,6 +270,31 @@ impl<'a> From<Protocol<'a>> for Multiaddr {
     }
 }
 
+impl From<SocketAddr> for Multiaddr {
+    fn from(v: SocketAddr) -> Multiaddr {
+        match v {
+            SocketAddr::V4(a) => a.into(),
+            SocketAddr::V6(a) => a.into()
+        }
+    }
+}
+
+impl From<SocketAddrV4> for Multiaddr {
+    fn from(v: SocketAddrV4) -> Multiaddr {
+        let mut addr: Multiaddr = v.ip().clone().into();
+        addr.push(Protocol::Tcp(v.port()));
```

Comment:
Might as well be `Protocol::Udp`.

Reply:
Ah of course - so this conversion won't work because we don't have enough information - ok I'll remove it